hopp/simulation/technologies/wind/wind_plant.py

```diff
@@ -191,17 +200,79 @@ def __attrs_post_init__(self):
 
         self._dispatch = None
 
-        self.turb_rating = self.config.turbine_rating_kw
+        if self.config.turbine_rating_kw is not None:
+            self.turb_rating = self.config.turbine_rating_kw
         self.num_turbines = self.config.num_turbines
+            
+        if self.config.model_name=="pysam":
+            self.initialize_pysam_wind_turbine()
+        
+        
+    def initialize_pysam_wind_turbine(self):
+        """Initialize wind turbine parameters for PySAM simulation.
+
+        Raises:
+            ValueError: if invalid turbine name is provided. Print list of valid turbine names before error is raised. 
+            UserWarning: discrepancy in rotor_diameter value
+            UserWarning: discrepancy in hub-height value
+        """
+
+        if self.config.model_name != "pysam":
+            return
```

Comment:
Either this logic or the check above it is unnecessary; suggest to remove one of them. The code can't get to this point unless the model_name is pysam already